.github/workflows/releases.yml

```
@@ -7,91 +7,113 @@ on:
+  schedule:
+  #  Every day at 00:00
+    - cron: '0 0 * * *'
+
+permissions:
+  contents: write
 
 jobs:
-  linux-release:
-    runs-on: ubuntu-20.04
+  prepare:
+    name: Prepare release
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.vars.outputs.branch }}
+      commit_sha: ${{ steps.sha.outputs.commit_sha }}
+      has_changes: ${{ steps.changes.outputs.has_changes }}
+      changelog: ${{ steps.changes.outputs.changelog }}
+      recent_commits: ${{ steps.changes.outputs.recent_commits }}
     steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-          ref: ${{ github.event.inputs.branch }}
-      
-      - name: Set env vars
-        id: envs
+      - name: Select branch
+        id: vars
         run: |
-          commit_sha=$(git rev-parse HEAD)
-          date=$(date +%Y_%m_%d)
-          echo "commit_sha=$commit_sha" >> $GITHUB_OUTPUT
-          echo "date=$date" >> $GITHUB_OUTPUT
-          echo "output_name=mrtrix3-linux-$commit_sha-$date" >> $GITHUB_OUTPUT
+          if [ "${{ github.event_name }}" = "workflow_dispatch" ]; then
+            echo "branch=${{ github.event.inputs.branch }}" >> "$GITHUB_OUTPUT"
+          else
+            # Nightly builds always from dev
+            echo "branch=dev" >> "$GITHUB_OUTPUT"
+          fi
 
-      - name: Install Eigen3
-        run: |
-          git clone https://gitlab.com/libeigen/eigen.git && cd eigen && git checkout 3.4.0
-          cmake -B build && cmake --build build
-          sudo cmake --install build
-
-      - name: Install Qt 6
-        uses: jurplel/install-qt-action@v3
-        with:
-          version: '6.7.0'
-          set-env: true
-
-      - name: Run build
-        run: |
-          ./packaging/package-linux-tarball.sh .
-          mv mrtrix.tar.gz ${{ steps.envs.outputs.output_name }}.tar.gz
-
-      - name: Upload release artifact
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ steps.envs.outputs.output_name }}
-          path: ${{ steps.envs.outputs.output_name }}.tar.gz
-
-  macos-release:
-    runs-on: macos-latest
-    steps:
-      - uses: actions/checkout@v3
+      - name: Checkout selected branch
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
-          ref: ${{ github.event.inputs.branch }}
+          ref: ${{ steps.vars.outputs.branch }}
+
+      - name: Capture commit SHA
+        id: sha
+        run: echo "commit_sha=$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
 
-      - name: Set env vars
-        id: envs
+      - name: Determine if there are new commits since last nightly
+        id: changes
         run: |
-          commit_sha=$(git rev-parse HEAD)
-          date=$(date +%Y_%m_%d)
-          echo "commit_sha=$commit_sha" >> $GITHUB_OUTPUT
-          echo "date=$date" >> $GITHUB_OUTPUT
-          echo "output_name=mrtrix3-macos-$commit_sha-$date" >> $GITHUB_OUTPUT
+          RECENT="$(git log -n 15 --oneline || true)"
```

```suggestion
          RECENT=$(git log --oneline | grep "Merge pull request #\d+ from MRtrix3\/\w+" | head -n15)
```

Amend as necessary; also subsequent code would need to reflect that this is a list of merged PRs rather than commits. But I think attaching to nightly releases a list of the most recent PR merges to `dev` that are included within would be more interpretable than just the most recent commits. As long as `dev` remains protected the build triggering shouldn't be affected.